xrviz/dashboard.py

```diff
@@ -57,42 +62,65 @@ def set_data(self, data):
         else:
             raise ValueError
 
+    def set_coords(self, *args):
+        # We can't reset indexed coordinates so add them every time
+        # in coord_selector.value
+        self.data = self.data.reset_coords()
+        indexed_coords = set(self.data.dims) & set(self.data.coords)
+        new_coords = list(set(self.control.coord_setter.coord_selector.value) | set(indexed_coords))
+        self.data = self.data.set_coords(new_coords)
+        self.control.set_coords(self.data)
+
     def create_plot(self, *args):
         self.kwargs = self.control.kwargs
         self.var = self.kwargs['Variables']
-        if self.is_dataset:
-            self.var_dims = list(self.data[self.var].dims)
-        else:
-            self.var_dims = list(self.data.dims)
-        #  var_selector_dims refers to dims for which index_selectors would be created
-        self.var_selector_dims = sorted([dim for dim in self.var_dims if dim not in [self.kwargs['x'], self.kwargs['y']]])
-
         self.index_selectors = []
         self.output[1].clear()  # clears Index_selectors
         self.output[2].clear()  # clears Players
 
         if self.is_dataset:
-            for dim in self.var_selector_dims:
-                selector = pn.widgets.Select(name=dim, options=list(self.data[self.var][dim].values))
-                self.index_selectors.append(selector)
-                selector.param.watch(self.callback_for_indexed_graph, ['value'], onlychanged=False)
-        else:
+            x = self.kwargs['x']
+            if not self.is_non_indexed_coord(x):  # i.e is a var_dim
```

Comment:
This case `# i.e is a var_dim` and `# if is_dataArray`, are almost same, except a few lines in between. Initially they were combined (for the lines which are same). Now I have separated them, to accomplish the task here. Please let me know if there is a better way of arranging these three cases here.

Reply:
I suspect indeed this could be cleaner, or simply moved into (testable) functions in utils. However, functionality trumps code cleanliness, so let's first ensure that the behaviour is as required.